packages/expo-maps/ios/AppleMapsViewState.swift

```diff
@@ -0,0 +1,11 @@
+// Copyright 2025-present 650 Industries. All rights reserved.
+
+import ExpoModulesCore
+import MapKit
+import SwiftUI
+
+@available(iOS 18.0, *)
+public class AppleMapsViewState: ObservableObject {
+  @Published var mapCameraPosition: MapCameraPosition = .automatic
+  @Published var selection: MapSelection<MKMapItem>?
+}
```

Comment:
```suggestion
import SwiftUI
import Observation

@available(iOS 18.0, *)
@Observable
public class AppleMapsViewState {
  var mapCameraPosition: MapCameraPosition = .automatic
  var selection: MapSelection<MKMapItem>?
}
```
We should now be able to use the Observation framework

Reply:
Unfortunately, I'm unable to get it to work. `The compiler is unable to type-check this expression in reasonable time; try breaking up the expression into distinct sub-expressions`

Reply:
@jakex7 interesting, ok we can look in to it. It doesn't need to block this PR